[WIP] Fixes generate-config in init-xmcp to only output a single path key.

joelachance opened the pull request:
## Summary

Fixes generate-config in init-xmcp to only output a single path key.

## Type of Change

- [X] Bug fixing
- [ ] Adding a feature
- [ ] Improving documentation
- [ ] Adding or updating examples
- [ ] Performance - bundle size improvement (if applicable)

## Affected Packages

- [ ] `xmcp` (core framework)
- [ ] `create-xmcp-app`
- [X] `init-xmcp`
- [ ] Documentation
- [ ] Examples

## Screenshots/Examples

New output: 

```javascript
import { type XmcpConfig } from "xmcp";

const config: XmcpConfig = {
  http: true,
  experimental: {
    adapter: "express",
  },
  paths: {
    tools: "src/tools",
    prompts: "src/prompts",
  },
};

export default config;
```

## Related Issues

Fixes #144

valebearzotti reviewed (changes requested):
It should actually support all three scenarios

Only tools: paths: { tools: "..." } // prompts set to false
Only prompts: paths: { prompts: "..." } // tools set to false
Both: paths: { tools: "...", prompts: "..." }

why? ideally, like we're doing for create-xmcp-app, you should be able to prompt if you want to initialize your project with both paths or only one of them (another issue, related but not this one in particular) so this leaves room for that support!

and there's an extra closing }, being added here:
```
configContent += `
  },`;
  ```

joelachance commented:
checking on the extra closing bracket!

it wasn't clear to me going through init-xmcp if you could skip either the tools or prompts folder. If you enter through all questions in the setup you get both /tools and /prompts out of the box.

Sounds like that's not supported yet though? So right now you have to have both tools and prompts dirs?

joelachance commented:
Ok, I fixed the brace, see output below (in the scenario we have both tools and prompts paths). I've also added tests, because I might be misunderstanding what we need here, but based on what you noted above:

> Only tools: paths: { tools: "..." } // prompts set to false
Only prompts: paths: { prompts: "..." } // tools set to false
Both: paths: { tools: "...", prompts: "..." }

This should work. You can log out the configContext for each test and see the results. Nothing's been working all day though, so 🤷‍♂️. 

I updated the PR to WIP in case I'm still missing something here.

```javascript
import { type XmcpConfig } from "xmcp";

const config: XmcpConfig = {
  http: true,
  experimental: {
    adapter: "nextjs",
  },
  paths: {
    tools: "./tools",
    prompts: "./prompts",
  },
};

export default config;
```
